csr_check.go

```diff
@@ -128,12 +138,21 @@ func authorizeCSR(config ClusterMachineApproverConfig, machines []v1beta1.Machin
 		return authorizeNodeClientCSR(config, machines, nodes, req, csr)
 	}
 
-	// node serving cert validation after this point
+	if isNodeServingCert(req, csr) {
+		return authorizeNodeServingCSR(machines, req, csr)
+	}
+
+	klog.Infof("CSR %q is not a client/serving CSR", req.Name)
+	// we don't want to requeue if none of the above was met so we return nil.
+	return nil
```

Comment:
The PR describes the changes as a refactor, but it looks like there's a change in behavior here.  Previously, an error would have been returned if it wasn't a client CSR, and didn't match the expected format of a server CSR.  The changes seem fine, but would you like to update the description?

Reply:
I'm actually keeping current error returning to keep unit test coverage and minimise scope of the PR.
I'll do farther refinements as small follow ups